volume/volume.go

```diff
@@ -73,6 +73,12 @@ type LabeledVolume interface {
 	Volume
 }
 
+// OptionedVolume wraps a Volume with driver defined options
+type OptionedVolume interface {
```

Comment:
Thanks @cpuguy83. I updated the PR and use `VolumeWithDetails` for now. Let me know if other names are preferred.